x-pack/plugins/index_management/server/routes/api/inference_models/register_get_route.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { InferenceAPIConfigResponse } from '@kbn/ml-plugin/common/types/trained_models';
```

Comment:
It'd be great if we can move this type to the `x-pack/packages/ml/trained_models_utils` package so we don't add the ML plugin as a dependency just to import a single type.

Reply:
@sphilipse Does it make sense use [any](https://github.com/elastic/kibana/pull/179575/files#diff-6a7bed4d58950f7e3523b1dfbc7fe8877430b9ecc3d15e102be1cc72158f6864R24) instead of InferenceAPIConfigResponse? In that case, we do not need to add ML plugin as a dependency just to import a single type. 🤔

Reply:
No, we shouldn't use `any` if we can at all avoid it. We know what the response shape is here so we should type it that way.

Reply:
Using InferenceAPIConfigResponse type from '@kbn/ml-plugin/common/types/trained_models'. We are going to use ml plugin for gathering information about trained models in upcoming PRs. So, I think it should be fine to add ML plugin as a dependency here